tests/test_raise_deprecation_warning.py

```diff
@@ -0,0 +1,58 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import importlib
+import unittest
+from unittest import mock
+
+from parameterized import parameterized
+
+
+class TestRaiseDeprecationWarning(unittest.TestCase):
+    PATH_TO_CLASSES = [
+        (
+            "airflow.gcp.hooks.compute.ComputeEngineHook",
+            "airflow.contrib.hooks.gcp_compute_hook.GceHook",
+        ),
+    ]
+
+    def assert_is_subclass(self, sub_class, parent_class):
+        self.assertTrue(issubclass(sub_class, parent_class),
+                        "{} is not subclass of {}".format(parent_class, sub_class))
+
+    def get_class_from_path(self, path_to_class):
+        path, _, class_name = path_to_class.rpartition(".")
+        module = importlib.import_module(path)
+        class_ = getattr(module, class_name)
+        return class_
+
+    @parameterized.expand(PATH_TO_CLASSES)
+    def test_if_class_throw_deprecation_warning(self, new_module, old_module):
+        old_module_class = self.get_class_from_path(old_module)
+        with self.assertWarns(DeprecationWarning):
+            with mock.patch("{}.__init__".format(new_module)):
+                old_module_class()
```

Comment:
```suggestion
                old_module_class()
                init_mock.assert_called_once_with()
```
I think it’s worth to check if the mock was called.